### PR TITLE
add a release binary cross-compiler builder container for Linux 

### PR DESCRIPTION
**Pull request by qrkourier**
resolves https://github.com/openziti/ziti-cmake/issues/3

### CODE REVIEW

**Comment by ekoby** [comment]
I'd use this image for other projects, can we rename it -- drop tunnel-sdk -- `ziti-c-builder` seems like a more generic name?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,21 @@
...
+# builder container for ziti-edge-tunnel
+
+This container image builds the `ziti-edge-tunnel` binary for Linux from source with statically-linked crypto libraries. This container image is available from Docker Hub as [openziti/ziti-tunnel-sdk-c-builder](https://hub.docker.com/r/openziti/ziti-tunnel-sdk-c-builder).
+
+## Build ziti-edge-tunnel
+
+To build the ziti-edge-tunnel binary, run the following command in the root of the **ziti-tunnel-sdk-c** repository:
+
+```bash
+docker run --rm --volume "${PWD}:/github/workspace" openziti/ziti-tunnel-sdk-c-builder
```
Comment (ekoby) [inline README.md:10]: this seems weird it I run the docker locally

File: README.md [hunk 2]
```diff
@@ -1,2 +1,21 @@
...
+```
+
+The default build target is "bundle" which produces a ZIP archive in `./build/package`. The executable binary is also stored in `./build/programs/ziti-edge-tunnel/Release/ziti-edge-tunnel`.
```
Comment (ekoby) [inline README.md:13]: I did not see a way to override default target

File: README.md (hunk 1 above)
Comment (qrkourier) [reply README.md:10]: Do you mean you got an unexpected result?

File: docker-image/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+set -euo pipefail
+
+# these commands must be in the entrypoint so they are run after workspace is mounted on Docker workdir
+echo "INFO: GIT_DISCOVERY_ACROSS_FILESYSTEM=${GIT_DISCOVERY_ACROSS_FILESYSTEM}"
+echo "INFO: WORKDIR=${PWD}"
+echo "INFO: $(git --version)"
+
+# if first positional is an expected arch string then set cmake preset,
+# else use ci-linux-x64 (which actually just uses native/host tools - e.g. not cross compile)
+if [ ${#} -ge 1 ]; then
```
[the remark this answers is not in the document]
Comment (qrkourier) [reply docker-image/entrypoint.sh:15]: I meant to deliberately avoid designing an API to the build procedure with various options and arguments. Alternatives include:

1. A script that reflects the CI build procedure
2. Override with arbitrary CMake options and arguments

Here's an example of overriding the simple entrypoint script with arbitrary CMake ops and args. Does this meet the need and expectation?

```bash

docker run --rm --volume "${PWD}:/github/workspace" --entrypoint='' openziti/ziti-tunnel-sdk-c-builder \
cmake \
    -E make_directory \
    ./build

docker run --rm --volume "${PWD}:/github/workspace" --entrypoint='' openziti/ziti-tunnel-sdk-c-builder \
cmake \
    --preset "ci-linux-x64" \
    -DCMAKE_BUILD_TYPE="Release" \
    -DBUILD_DIST_PACKAGES=OFF \
    -DVCPKG_OVERLAY_PORTS="./vcpkg-overlays/linux-syslibs/ubuntu18" \
    -S . \
    -B ./build

docker run --rm --volume "${PWD}:/github/workspace" --entrypoint='' openziti/ziti-tunnel-sdk-c-builder \
cmake \
    --build ./build \
    --config "Release" \
    --target bundle \
    --verbose
```

**Comment by qrkourier** [comment]
> I'd use this image for other projects, can we rename it -- drop tunnel-sdk -- `ziti-c-builder` seems like a more generic name?

You wish to explore using a common container image for these four things, correct?

1. build C-SDK locally
2. build C-SDK in CI
3. build T-SDK locally
4. build T-SDK in CI

**Comment by ekoby** [comment]
> > I'd use this image for other projects, can we rename it -- drop tunnel-sdk -- `ziti-c-builder` seems like a more generic name?
> 
> You wish to explore using a common container image for these four things, correct?
> 
> 1. build C-SDK locally
> 2. build C-SDK in CI
> 3. build T-SDK locally
> 4. build T-SDK in CI

5. build `zitify` in CI
6. build `zitify` locally

the list may grow in the future

**Comment by qrkourier** [comment]
> 6\. build `zitify` locally

I verified I can build zitify-pack with this container by overriding the entrypoint script. Assuming this container image were renamed more generically, would a new shell script in the `zitify` repo like `build.sh` with these contents meet your need for building `zitify` locally?

```bash
docker run --rm --volume "${PWD}:/github/workspace" --entrypoint='' openziti/ziti-tunnel-sdk-c-builder \
          bash -c 'rm -rf /github/workspace/build \
          && cmake -E make_directory /github/workspace/build \
          && cmake -DCMAKE_TOOLCHAIN_FILE="" -S /github/workspace -B /github/workspace/build \
          && cmake --build /github/workspace/build \
          && cmake --build /github/workspace/build --target zitify-pack'
```

**Review by ekoby — APPROVED** [review]
works great! https://github.com/openziti/ziti-sdk-c/pull/520

**Pull request merged**